code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -852,198 +929,198 @@ pub mod pallet {
+					})?;
 
-			// REVIEW(benluelo): Make this logic a method on Stake
-			let stake_with_penalty = if is_early_unlock {
-				(Perbill::one() - stake.lock.unlock_penalty).mul_ceil(stake.stake)
-			} else {
-				stake.stake
-			};
+				// REVIEW(benluelo): Make this logic a method on Stake
+				let staked_amount_returned_to_staker = if is_early_unlock {
+					(Perbill::one() - stake.lock.unlock_penalty).mul_ceil(stake.amount)
+				} else {
+					stake.amount
+				};
 
-			let fnft_asset_account =
-				T::FinancialNft::asset_account(fnft_collection_id, fnft_instance_id);
+				let fnft_asset_account =
+					T::FinancialNft::asset_account(fnft_collection_id, fnft_instance_id);
 
-			T::Assets::remove_lock(T::LockId::get(), asset_id, &fnft_asset_account)?;
-			T::Assets::remove_lock(T::LockId::get(), share_asset_id, &fnft_asset_account)?;
-			T::Assets::transfer(
-				asset_id,
-				&fnft_asset_account,
-				&owner,
-				stake_with_penalty,
-				keep_alive,
-			)?;
-
-			Stakes::<T>::remove(fnft_collection_id, fnft_instance_id);
+				T::Assets::remove_lock(
+					T::LockId::get(),
+					stake.staked_asset_id,
+					&fnft_asset_account,
+				)?;
+				T::Assets::remove_lock(T::LockId::get(), share_asset_id, &fnft_asset_account)?;
+				T::Assets::transfer(
+					stake.staked_asset_id,
+					&fnft_asset_account,
+					&owner,
+					staked_amount_returned_to_staker,
+					keep_alive,
+				)?;
 
-			// Burn the financial NFT and the shares it holds
-			T::Assets::burn_from(asset_id, &fnft_asset_account, stake.stake - stake_with_penalty)?;
-			T::Assets::burn_from(share_asset_id, &fnft_asset_account, stake.stake)?;
-			T::FinancialNft::burn(fnft_collection_id, fnft_instance_id, Some(who))?;
+				// Burn the financial NFT and the shares it holds
+				if is_early_unlock {
+					// If there is no penalty then there is nothing to burn as it will all have been
+					// transferred back to the staker. burn_from isn't a noop if the amount to burn
+					// is 0, hence the check
+					T::Assets::burn_from(
+						stake.staked_asset_id,
+						&fnft_asset_account,
+						stake.amount.defensive_saturating_sub(staked_amount_returned_to_staker),
+					)?;
+				}
+				T::Assets::burn_from(share_asset_id, &fnft_asset_account, stake.amount)?;
+				T::FinancialNft::burn(fnft_collection_id, fnft_instance_id, Some(who))?;
 
-			Self::deposit_event(Event::<T>::Unstaked {
-				owner: who.clone(),
-				fnft_collection_id: *fnft_collection_id,
-				fnft_instance_id: *fnft_instance_id,
-			});
+				Self::deposit_event(Event::<T>::Unstaked {
+					owner: who.clone(),
+					fnft_collection_id: *fnft_collection_id,
+					fnft_instance_id: *fnft_instance_id,
+				});
 
-			Ok(())
+				Ok(())
+			})
 		}
 
 		#[transactional]
 		fn split(
 			who: &Self::AccountId,
-			(fnft_collection_id, fnft_instance_id): &Self::PositionId,
+			(fnft_collection_id, old_fnft_instance_id): &Self::PositionId,
+			// REVIEW(benluelo): Rename this argument so it's clearer what the split is doing;
```

Comment:
I would not change this, it just means "split by the given ratio". There is no keep or remove here. Both kept and owned act the same other than the size "ratio"

Reply:
perhaps this is just me, but it's not super clear which position gets which part of the ratio (without looking into the implementation). I have no problem leaving it as-is, I only left the comment to spark discussion about it and see if we could come up with a name that described the functionality better

Reply:
Which position gets which ratio is not important from business logic as they are both owned by the same user. There is no advantage to existing position being the smaller or the large position out of the two.

> I have no problem leaving it as-is, I only left the comment to spark discussion

I understand, that's good :) , but let's not leave the comments without resolution in the code as that would confuse anyone else looking at the code.

Reply:
> they are both owned by the same user
good point, I guess it doesn't really matter :smile:

I'll remove the comment :+1: